src/prof.c

```diff
@@ -778,7 +783,7 @@ prof_tdata_cleanup(tsd_t *tsd) {
 		return;
 	}
 
-	tdata = tsd_prof_tdata_get(tsd);
```

Comment:
Any reason to change this? This is the shutdown path; using the tsd getter might be right.

Reply:
`prof_tdata_get(tsd, false)` is identical to `tsd_prof_tdata_get(tsd)`. I'm trying to make the code consistent, and also make it look more explicit that we're not creating `tdata` here.

Reply:
I think the point here is slightly different. On the shutdown path, we should stick to the low level getter to retrieve the underlying data. Right now they may be identical; however it won’t be crazy for us to say, add a tsd_nominal guard in the prof getter. On the cleanup path here, it would be safer to assume things are not working as normal.

Reply:
See a similar example for tcache https://github.com/jemalloc/jemalloc/blob/dev/src/tcache.c#L650
where the higher level tcache_get won’t work on the shutdown path, and the tsd getter is used directly.